File: fe_admin/admin.quadra_mall_web-main/src/types/itemType.ts
export interface ItemType {
    id: number; // Bắt buộc vì backend luôn trả id
    name: string;
    slug: string;
    description?: string;
    iconUrl?: string;
    isActive: boolean;
    parent?: ItemType;
    children?: ItemType[];
}

export type CreateItemTypePayload = {
    name: string;
    description?: string;
    iconUrl?: string;
    isActive: boolean;
    parent?: { id: number };
};

File: be/quadramall_api-main/src/main/java/com/quadra/ecommerce_api/entity/store/ItemType.java
package com.quadra.ecommerce_api.entity.store;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder

@Entity
@Table(name = "item_types")
public class ItemType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "parent_id")
    private ItemType parent;

    @Column(nullable = false)
    private String name;

    @Column(unique = true, nullable = false)
    private String slug;

    @Column
    private String description;

    @Column(name = "icon_url")
    private String iconUrl;

    @Builder.Default
    @Column(name = "is_active", nullable = false)
    private boolean isActive = true;
}
